# **Sentiment Analysis with NLTK**

<h2 style="font-family: 'poppins'; font-weight: bold; color: Green;">👨💻 Author: Irfan Ullah Khan</h2>

[![GitHub](https://img.shields.io/badge/GitHub-Profile-blue?style=for-the-badge&logo=github)](https://github.com/programmarself)
[![Kaggle](https://img.shields.io/badge/Kaggle-Profile-blue?style=for-the-badge&logo=kaggle)](https://www.kaggle.com/programmarself)
[![LinkedIn](https://img.shields.io/badge/LinkedIn-Profile-blue?style=for-the-badge&logo=linkedin)](https://www.linkedin.com/in/irfan-ullah-khan-4a2871208/)  

[![YouTube](https://img.shields.io/badge/YouTube-Profile-red?style=for-the-badge&logo=youtube)](https://www.youtube.com/@irfanullahkhan7748)
[![Email](https://img.shields.io/badge/Email-Contact%20Me-red?style=for-the-badge&logo=email)](mailto:programmarself@gmail.com)
[![Website](https://img.shields.io/badge/Website-Contact%20Me-red?style=for-the-badge&logo=website)](https://flowcv.me/ikm)


This project uses the Natural Language Toolkit (NLTK) to perform sentiment analysis on text data.



In [1]:
# Install NLTK
!pip install nltk

In [2]:
# Import Libraries
import nltk
from nltk.sentiment.vader import SentimentIntensityAnalyzer
from nltk.tokenize import word_tokenize
nltk.download('vader_lexicon')
nltk.download('punkt')

[nltk_data] Downloading package vader_lexicon to /root/nltk_data...
[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt.zip.


True

In [3]:
# Initialize the sentiment analyzer
sid = SentimentIntensityAnalyzer()

In [6]:
# Sample text
text = "I love using Google Colab for python projects. It is fantastic!"

In [7]:
# Perform sentiment analysis
scores = sid.polarity_scores(text)
print(scores)

{'neg': 0.0, 'neu': 0.497, 'pos': 0.503, 'compound': 0.8439}


**1):  'neg' (Negative):** The proportion of the text that is considered negative. In this case, 0.0 means there is no negative sentiment detected in the text.

**2): 'neu' (Neutral):** The proportion of the text that is considered neutral. Here, 0.527 indicates that 52.7% of the text is neutral, meaning it's neither positive nor negative.

**3): 'pos' (Positive):** The proportion of the text that is considered positive. A value of 0.473 means that 47.3% of the text is positive.

**4): 'compound':** The compound score is a normalized score that combines all the individual sentiment scores into a single value.
It ranges from -1 (most negative) to +1 (most positive). In this case, 0.8439 indicates a strongly positive sentiment.


**Overall Sentiment:**\
 The compound score of 0.8439 suggests that the overall sentiment of the text is positive.\
**Distribution of Sentiments:**\
0% of the text is negative.\
49.7% is neutral.\
50.3% is positive.